shell/platform/darwin/macos/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -372,6 +436,35 @@ - (void)updateTextAndSelection {
   }
 }
 
+- (NSString*)textAffinityString {
+  return (self.textAffinity == FlutterTextAffinityUpstream) ? kTextAffinityUpstream
+                                                            : kTextAffinityDownstream;
+}
+
+- (NSRange)clampSelection:(NSRange*)range forText:(NSString*)text {
+  int start = MIN(MAX(range->location, 0), text.length);
+  int length = MIN(range->length, text.length - start);
+  return NSMakeRange(start, length);
+}
+
+// Change the range of selected text, without notifying the framework.
+- (void)setSelectedTextRangeLocal:(NSRange*)selectedTextRange {
```

Comment:
Is this method being called anywhere?

Reply:
Looks like it's not on Mac, good catch.  Removing.